storm-client/src/jvm/org/apache/storm/windowing/WaterMarkEventGenerator.java

```diff
@@ -45,17 +46,25 @@
     private final Map<GlobalStreamId, Long> streamToTs;
     private final ScheduledExecutorService executorService;
     private final int interval;
+    private final AtomicLong lastWaterMarkTs;
```

Comment:
Yes. Why do you prefer volatile? (I'll be happy to change it, I'm just wondering if there's a drawback to using AtomicX)